File: 2023-11-15-multiple-sns-login-with-spring-security-oauth2-client/action-in-blog/src/main/java/action/in/blog/domain/entity/UserEntity.java
package action.in.blog.domain.entity;

import action.in.blog.domain.enums.LoginType;
import action.in.blog.domain.enums.Role;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "TB_USER")
public class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @Enumerated(value = EnumType.STRING)
    private Role role;
    private String email;
    private String nickName;
    @Enumerated(value = EnumType.STRING)
    private LoginType loginType;
    @Column(name = "oauth2_client_id")
    private String oauth2ClientId;
}

File: 2023-08-13-jpa-entity-serialize-exception-with-redis-session/action-in-blog/src/main/kotlin/action/in/blog/domain/entity/UserEntity.kt
package action.`in`.blog.domain.entity

import jakarta.persistence.*

@Entity
@Table(name = "tb_user")
class UserEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,
    val name: String,
    @ElementCollection
    @CollectionTable(name = "tb_favorite_posts")
    val favoritePosts: MutableList<Long>
)